apm-agent-core/src/test/java/co/elastic/apm/agent/bci/InstrumentationTest.java

```diff
@@ -104,7 +105,12 @@ void testIntercept() {
         assertThat(interceptMe()).isEqualTo("intercepted");
     }
 
+    /*
+     * todo: this test is invalid because the agent CL here is the App CL, whereas in reality it would be the bootstrap CL
```

Comment:
Is that test failing now or does it just not test what it should test?

Reply:
It is failing now because I [didn't allow](https://github.com/elastic/apm-agent-java/pull/1660/files#diff-2ec63f545539ff7d65f52ef01087a18c7cdd9eb7130de156fb412ae6fe9e580eR324) not having the agent jar. I can make it less strict, but I don't like having testing-specific logic in the production code...

Reply:
👌